src/Cake.Common/Tools/NuGet/NuGetMSBuildVersion.cs

```diff
@@ -27,6 +27,11 @@ public enum NuGetMSBuildVersion
         /// <summary>
         /// MSBuildVersion : <c>15</c>
         /// </summary>
-        MSBuild15 = 15
+        MSBuild15 = 15,
+
+        /// <summary>
+        /// MSBuildVersion : <c>16</c>
+        /// </summary>
+        MSBuild16 = 16
```

Comment:
Hi,

Thanks for submitting this pull request 👍 

After feedback from the NuGet team that `16` isn't supported, but `16.0` is, actually the same for `15`. 
Valid values are 4, 12, 14, 15.1, 15.3, 15.4, 15.5, 15.6, 15.7, 15.8, 15.9, 16.0 (_yes there's no 15.2_ 😉 )
As decimals isn't supported by enums i propose you adjust the enum to something like this

```csharp
    /// <summary>
    /// NuGet MSBuild version
    /// </summary>
    public enum NuGetMSBuildVersion
    {
        /// <summary>
        /// MSBuildVersion : <c>4</c>
        /// </summary>
        MSBuild4 = 4,

        /// <summary>
        /// MSBuildVersion : <c>12</c>
        /// </summary>
        MSBuild12 = 12,

        /// <summary>
        /// MSBuildVersion : <c>14</c>
        /// </summary>
        MSBuild14 = 14,

        /// <summary>
        /// MSBuildVersion : <c>15</c>
        /// </summary>
        /// <remarks>
        /// MSBuildVersion 15 isn't an valid option and will be removed in the futute.
        /// </remarks>
        [Obsolete("MSBuildVersion 15 isn't an valid option and will be removed in the future.")]
        MSBuild15 = 15,

        /// <summary>
        /// MSBuildVersion : <c>15.1</c>
        /// </summary>
        MSBuild15_1 = 1501,

        /// <summary>
        /// MSBuildVersion : <c>15.3</c>
        /// </summary>
        MSBuild15_3 = 1503,

        /// <summary>
        /// MSBuildVersion : <c>15.4</c>
        /// </summary>
        MSBuild15_4 = 1504,

        /// <summary>
        /// MSBuildVersion : <c>15.5</c>
        /// </summary>
        MSBuild15_5 = 1505,

        /// <summary>
        /// MSBuildVersion : <c>15.6</c>
        /// </summary>
        MSBuild15_6 = 1506,

        /// <summary>
        /// MSBuildVersion : <c>15.7</c>
        /// </summary>
        MSBuild15_7 = 1507,

        /// <summary>
        /// MSBuildVersion : <c>15.8</c>
        /// </summary>
        MSBuild15_8 = 1508,

        /// <summary>
        /// MSBuildVersion : <c>15.9</c>
        /// </summary>
        MSBuild15_9 = 1509,

        /// <summary>
        /// MSBuildVersion : <c>16.0</c>
        /// </summary>
        MSBuild16_0 = 1600
    }
```

Then add a `NuGetMSBuildVersionExtensions.cs` with helper extension method that switches on the enum and returns the appropriate string, something like

```csharp
    public static string GetNuGetMSBuildVersionString(this NuGetMSBuildVersion nuGetMSBuildVersion)
    {
        switch (nuGetMSBuildVersion)
        {
            case NuGetMSBuildVersion.MSBuild4:
                return "4";

            case NuGetMSBuildVersion.MSBuild12:
                return "12";

            case NuGetMSBuildVersion.MSBuild14:
                return "14";

            case NuGetMSBuildVersion.MSBuild15:
                return "15";

            case NuGetMSBuildVersion.MSBuild15_1:
                return "15.1";

            case NuGetMSBuildVersion.MSBuild15_3:
                return "15.3";

            case NuGetMSBuildVersion.MSBuild15_4:
                return "15.4";

            case NuGetMSBuildVersion.MSBuild15_5:
                return "15.5";

            case NuGetMSBuildVersion.MSBuild15_6:
                return "15.6";

            case NuGetMSBuildVersion.MSBuild15_7:
                return "15.7";

            case NuGetMSBuildVersion.MSBuild15_8:
                return "15.8";

            case NuGetMSBuildVersion.MSBuild15_9:
                return "15.9";

            case NuGetMSBuildVersion.MSBuild16_0:
                return "16.0";

            default:
                throw new ArgumentOutOfRangeException(nameof(nuGetMSBuildVersion), nuGetMSBuildVersion, "Invalid nuGetMSBuildVersion supplied.");
        }
    }
```


Changing all `builder.Append(settings.MSBuildVersion.Value.ToString("D"));` to something like `builder.Append(settings.MSBuildVersion.Value.GetNuGetMSBuildVersionString());`

Quick search reveals at least 3 places
https://github.com/cake-build/cake/blob/044fa351bec57403ed48cd0d545b8cc1a1f4309d/src/Cake.Common/Tools/NuGet/Pack/NuGetPacker.cs#L204
https://github.com/cake-build/cake/blob/a1131b748b539f81f3a8817ef00d71b32eaff630/src/Cake.Common/Tools/NuGet/Restore/NuGetRestorer.cs#L121
https://github.com/cake-build/cake/blob/a1131b748b539f81f3a8817ef00d71b32eaff630/src/Cake.Common/Tools/NuGet/Update/NuGetUpdater.cs#L99

This would add support for the correct values, while still being compatible with existing scripts.

Reply:
Wow, thanks for good advices.